common/sagas/contracts.js

```diff
@@ -0,0 +1,61 @@
+import { takeEvery, select, put } from 'redux-saga/effects';
+import type { Effect } from 'redux-saga/effects';
+import translate from 'translations';
+
+import {
+  setNodeContracts,
+  AccessContractAction,
+  setInteractiveContract
+} from 'actions/contracts';
+import { showNotification } from 'actions/notifications';
+import {
+  FETCH_NODE_CONTRACTS,
+  ACCESS_CONTRACT
+} from 'actions/contractsConstants';
+import { NODE_CHANGE } from 'actions/configConstants';
+import CONTRACTS from 'config/contracts';
+import { NODES } from 'config/data';
+import { getNode } from 'selectors/config';
+import { isValidETHAddress } from 'libs/validators';
+
+function* handleFetchNodeContracts() {
+  const nodeKey = yield select(getNode);
+  const node = NODES[nodeKey] || {};
+  const contracts = CONTRACTS[node.network] || [];
+  yield put(setNodeContracts(contracts));
```

Comment:
By state, do you mean the reducer store instead of component state? If so, I wanted to keep this logic separated out. I figured this would make the pre-select list more flexible for if we add local storage / some way of having nodes provide their own suggested contracts.

If this seems like too much overhead, I could move it into a mapStateToProps function and do all this there.